lib/ansible/plugins/inventory/__init__.py

```diff
@@ -453,8 +447,18 @@ def _add_host_to_keyed_groups(self, keys, variables, host, strict=False, fetch_h
                                 sep = ''
                             gname = self._sanitize_group_name('%s%s%s' % (prefix, sep, bare_name))
                             result_gname = self.inventory.add_group(gname)
+                            # Add `ansible_keyed_group_name` to the variables that we can use to template
+                            templar_variables = combine_vars({'ansible_keyed_group_name': result_gname}, variables)
```

Comment:
reverse the precedence or 'other vars' could overwrite  the internal variable

Reply:
Reversed the precedence in https://github.com/bcoca/ansible/pull/13/commits/df2db8380e7c44d427d1d3a52851532b1be1542c